l10n_it_ricevute_bancarie/models/account.py

```diff
@@ -97,17 +97,19 @@ def create(self, vals):
 
     @api.onchange("partner_id", "invoice_payment_term_id", "move_type")
     def _onchange_riba_partner_bank_id(self):
-        if (
-            not self.riba_partner_bank_id
-            or self.riba_partner_bank_id not in self.partner_id.bank_ids
+        if not self.riba_partner_bank_id or self.riba_partner_bank_id not in (
+            self.partner_id.bank_ids | self.partner_id.commercial_partner_id.bank_ids
```

Comment:
Questi conti si potrebbero mettere in una variabile tipo `allowed_banks` da usare sia qui che sotto, che ne pensi?

Reply:
volevo tenere la diff più piccola possibile, ma hai pienamente ragione. Ho riscritto il metodo per chiarezza.

Reply:
Ah ecco, com'è adesso in effetti il `diff` non è banale, e non mi è nemmeno chiarissimo se sia equivalente a quello che c'era prima; non si potrebbe fare così?
```diff
     @api.onchange("partner_id", "invoice_payment_term_id", "move_type")
     def _onchange_riba_partner_bank_id(self):
+        allowed_banks = (
+            self.partner_id.bank_ids or self.partner_id.commercial_partner_id.bank_ids
+        )
         if (
             not self.riba_partner_bank_id
-            or self.riba_partner_bank_id not in self.partner_id.bank_ids
+            or self.riba_partner_bank_id not in allowed_banks
         ):
             bank_ids = self.env["res.partner.bank"]
             if (
                 self.partner_id
                 and self.invoice_payment_term_id.riba
                 and self.move_type in ["out_invoice", "out_refund"]
             ):
-                bank_ids = self.partner_id.bank_ids
+                bank_ids = allowed_banks
             self.riba_partner_bank_id = bank_ids[0] if bank_ids else None
```
così hai le banche in una variabile e, oltre alla nuova variabile, cambi solo due righe del metodo; così il `diff` rimane piccolo ed è chiaro che la logica rimane la stessa (a parte il fallback aggiunto).

Reply:
@SirAionTech se ci dai l'ok la spariamo nell'iperspazio 🚀

Reply:
:+1: